backside/spots/domain/spot_db.go

```diff
@@ -2,29 +2,69 @@ package domain
 
 import (
 	"fmt"
-	"os"
-
 	"gorm.io/driver/postgres"
 	"gorm.io/gorm"
+	"os"
 	"sk8.town/backside/errs"
 	"sk8.town/backside/logger"
 )
 
 //go:generate mockgen --build_flags=--mod=mod -destination=../mocks/spot_repository.go -package=mocks sk8.town/backside/spots/domain SpotRepository
 type SpotRepository interface {
 	Add(*Spot) (*Spot, *errs.AppError)
+	Get(int) (*Spot, *errs.AppError)
+	GetAll() ([]*Spot, *errs.AppError)
+	Update(int, *Spot) (*Spot, *errs.AppError)
+	Delete(int) *errs.AppError
 }
 
 type SpotDb struct {
 	client *gorm.DB
 }
 
 func (db SpotDb) Add(spot *Spot) (*Spot, *errs.AppError) {
-	db.client.Create(spot)
+	err := db.client.Create(spot).Error
+	if err != nil {
+		return nil, errs.NewUnexpectedError(err.Error())
+	}
 	db.client.Find(&spot)
 	return spot, nil
 }
 
+func (db SpotDb) Get(id int) (*Spot, *errs.AppError) {
+	var spot Spot
+	err := db.client.Model(&Spot{}).Where("id = ?", id).Take(&spot).Error
+	if err != nil {
+		return nil, errs.NewUnexpectedError(err.Error())
+	}
+	return &spot, nil
+}
+
+func (db SpotDb) GetAll() ([]*Spot, *errs.AppError) {
+	var spots []*Spot
+	err := db.client.Model(&Spot{}).Find(&spots).Error
+	if err != nil {
+		return nil, errs.NewUnexpectedError(err.Error())
+	}
+	return spots, nil
+}
+
+func (db SpotDb) Update(id int, spotToUpdateWith *Spot) (*Spot, *errs.AppError) {
+	err := db.client.Model(&Spot{}).Where("id = ?", id).Take(&Spot{}).UpdateColumns(spotToUpdateWith.toMap()).Error
+	if err != nil {
+		return nil, errs.NewUnexpectedError(err.Error())
```

Comment:
Something to consider here is to hide all the database errors on a production environment due to security. It doesn't matter that much since this is still an open-source project, but we can do it for fun.

I would do it inside the error, like:
```go
if config.GetEnv() == config.Env.Production {
  message = "Internal server error" // overwrite
}
```